### ISSUE TITLE
reinforcement and lock not working on coke oven door

### ISSUE THREAD

**Issue by ritter-manni** [anegostudios/VintageStory-Issues#2282]
### Game Version

1.17.9

### Platform

Windows

### Modded

Vanilla

### SP/MP

Multiplayer

### Description

1.17.9 coke oven door reinforced with a steel ingot and a steel lock can be opened by anyone and wen opened it loses it reinforcement

### How to reproduce

1. reinforce the coke oven door (i used a steel ingot) 
2. place a lock on it (i used a steel lock)
3. let it be opend by anyone
4. the reinforcement and lock are gone

### Logs

```Log```

**Comment by noelle-lavenza** [comment]
This may be due to the coke oven door using `SetBlock` instead of `ExchangeBlock` like normal (pre-1.18) doors do. `SetBlock` causes the reinforcement data to be wiped for the block while `ExchangeBlock` leaves it intact, specifically because the former calls `BehaviorReinforcable.OnBlockPlaced` which calls `ModSystemBlockReinforcement.ClearReinforcement(pos)`. `ExchangeBlock` skips this and just swaps out the IDs and triggers updates.

Proposed solution: On these two lines, replace SetBlock with ExchangeBlock.
https://github.com/anegostudios/vssurvivalmod/blob/08fadd58f5bd39d5dd80fd5755f9534f2a02f12f/Block/BlockCokeOvenDoor.cs#L22
https://github.com/anegostudios/vssurvivalmod/blob/08fadd58f5bd39d5dd80fd5755f9534f2a02f12f/Block/BlockCokeOvenDoor.cs#L27
```diff

            if (Variant["state"] == "closed")
            {
-               world.BlockAccessor.SetBlock(world.GetBlock(CodeWithVariant("state", "opened")).Id, pos);
+               world.BlockAccessor.ExchangeBlock(world.GetBlock(CodeWithVariant("state", "opened")).Id, pos);
                if (world.Side == EnumAppSide.Server) world.BlockAccessor.TriggerNeighbourBlockUpdate(pos);
                world.PlaySoundAt(new AssetLocation("sounds/block/cokeovendoor-open"), pos.X + 0.5, pos.Y + 0.5, pos.Z + 0.5, byPlayer, true);
            } else
            {
-               world.BlockAccessor.SetBlock(world.GetBlock(CodeWithVariant("state", "closed")).Id, pos);
+               world.BlockAccessor.ExchangeBlock(world.GetBlock(CodeWithVariant("state", "closed")).Id, pos);
                if (world.Side == EnumAppSide.Server) world.BlockAccessor.TriggerNeighbourBlockUpdate(pos);
                world.PlaySoundAt(new AssetLocation("sounds/block/cokeovendoor-close"), pos.X + 0.5, pos.Y + 0.5, pos.Z + 0.5, byPlayer, true);
            }
```

**Comment by radfast** [comment]
Fixed in upcoming 1.18.2, thank you for reporting - and sorry it took such a long time to get to this!
